src/main/java/uk/gov/hmcts/darts/cases/service/impl/CloseOldCasesProcessorImpl.java

```diff
@@ -93,6 +93,8 @@ public void closeCase(Integer courtCaseId, UserAccountEntity userAccount) {
             for (HearingEntity hearingEntity : courtCase.getHearings()) {
                 eventList.addAll(hearingEntity.getEvents());
             }
+            // remove non current events
+            eventList.removeIf(event -> !event.isCurrent());
```

Comment:
Using Streams is usually preferred for this sort of thing . The best way to do this is to update the code in the for loop on line 93. 
```suggestion
      for (HearingEntity hearingEntity : courtCase.getHearings()) {
                eventList.addAll(hearingEntity.getEvents().stream()
                    .filter(EventEntity::isCurrent)
                    .toList());
            }
```

Reply:
This has been updated to use a stream